### PR TITLE
[DOC] Update title and description in README

### PR DESCRIPTION
**Pull request by kaijchen**
### What changes were proposed in this pull request?

Add "Apache" in the title and improve the description.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

See https://github.com/kaijchen/incubator-uniffle/tree/readme#apache-uniffle-incubating

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -15,10 +15,12 @@
   ~ limitations under the License.
   -->
 
-# What is Uniffle
+# Apache Uniffle (Incubating)
 
-Uniffle is a Remote Shuffle Service, and provides the capability for Apache Spark applications
-to store shuffle data on remote servers.
+Uniffle is a unified remote shuffle service for analytics engines.
```
Comment (jerqi) [inline README.md:20]: analytics engine?

File: README.md [hunk 2]
```diff
@@ -15,10 +15,12 @@
...
+It provides the ability to aggregate and store shuffle data on remote servers,
```
Comment (jerqi) [inline README.md:21]: aggregate?

File: README.md (hunk 1 above)
Comment (kaijchen) [reply README.md:20]: Spark says it's a "unified analytics engine for large-scale data processing", do you have any suggestions?

File: README.md (hunk 2 above)
Comment (kaijchen) [reply README.md:21]: I mean "batching writes". Any better word for this?

File: README.md (hunk 1 above)
Comment (jerqi) [reply README.md:20]: MapReduce seems not to be an analytics engine.

Comment (jerqi) [inline README.md:20]:
```suggestion
Uniffle is a unified remote shuffle service for compute engines.
```

File: README.md (hunk 2 above)
Comment (jerqi) [reply README.md:21]: buffer?

**Review by jerqi — APPROVED** [review]
LGTM

**Comment by kaijchen** [comment]
Thanks @jerqi for the review.

**Pull request merged**